packages/vision/src/v1/image_annotator_api.js

```diff
@@ -0,0 +1,188 @@
+ *
+ * @example
+ * var visionV1 = require('@google-cloud/vision').v1({
+ *   // optional auth parameters.
+ * });
+ * var api = visionV1.imageAnnotatorApi();
+ *
+ * @class
+ */
+function ImageAnnotatorApi(gaxGrpc, grpcClient, opts) {
+  opts = opts || {};
+  var servicePath = opts.servicePath || SERVICE_ADDRESS;
+  var port = opts.port || DEFAULT_SERVICE_PORT;
+  var sslCreds = opts.sslCreds || null;
+  var clientConfig = opts.clientConfig || {};
+  var timeout = opts.timeout || DEFAULT_TIMEOUT;
+  var appName = opts.appName || 'gax';
+  var appVersion = opts.appVersion || gax.Version;
+
+  var googleApiClient = [
+    appName + '/' + appVersion,
+    CODE_GEN_NAME_VERSION,
+    'nodejs/' + process.version].join(' ');
+
+  var defaults = gaxGrpc.constructSettings(
+      'google.cloud.vision.v1.ImageAnnotator',
+      configData,
+      clientConfig,
+      timeout,
+      null,
+      null,
+      {'x-goog-api-client': googleApiClient});
+
+  var stub = gaxGrpc.createStub(
+      servicePath,
+      port,
+      grpcClient.google.cloud.vision.v1.ImageAnnotator,
+      {sslCreds: sslCreds});
+  var methods = [
+    'batchAnnotateImages'
+  ];
+  methods.forEach(function(methodName) {
+    this['_' + methodName] = gax.createApiCall(
+        stub.then(function(stub) { return stub[methodName].bind(stub); }),
+        defaults[methodName]);
+  }.bind(this));
+}
+
+// Service calls
+
+/**
+ * Run image detection and annotation for a batch of images.
+ *
+ * @param {Object[]} requests
+ *   Individual image annotation requests for this batch.
+ *
+ *   This object should have the same structure as [AnnotateImageRequest]{@link AnnotateImageRequest}
+ * @param {gax.CallOptions=} options
+ *   Overrides the default settings for this call, e.g, timeout,
+ *   retries, etc.
+ * @param {function(?Error, ?Object)=} callback
+ *   The function which will be called with the result of the API call.
+ *
+ *   The second parameter to the callback is an object representing [BatchAnnotateImagesResponse]{@link BatchAnnotateImagesResponse}
+ * @returns {gax.EventEmitter} - the event emitter to handle the call
+ *   status.
+ *
+ * @example
+ *
+ * var api = visionV1.imageAnnotatorApi();
+ * var requests = [];
+ * api.batchAnnotateImages(requests, function(err, response) {
+ *     if (err) {
+ *         console.error(err);
+ *         return;
+ *     }
+ *     // doThingsWith(response)
+ * });
+ */
+ImageAnnotatorApi.prototype.batchAnnotateImages = function batchAnnotateImages() {
+  var args = arguejs({
+    requests: Array,
+    options: [gax.CallOptions],
+    callback: [Function]
+  }, arguments);
+  var req = {
+    requests: args.requests
+  };
+  return this._batchAnnotateImages(req, args.options, args.callback);
+};
+
+function ImageAnnotatorApiBuilder(gaxGrpc) {
+  if (!(this instanceof ImageAnnotatorApiBuilder)) {
+    return new ImageAnnotatorApiBuilder(gaxGrpc);
+  }
+
+  var grpcClient = gaxGrpc.load([{
+    root: require('google-proto-files')('..'),
```

Comment:
Yes, but a module should define its own dependencies instead of borrowing from another. That can lead to unexpected disappearances of modules if the middle layer changes its dependencies.